storage/client.go

```diff
@@ -0,0 +1,48 @@
+package storage
+
+import (
+	"fmt"
+	"net/url"
+
+	"release-manager/storage/util"
+
+	"github.com/nedpals/supabase-go"
+)
+
+const (
+	// signedURLExpiration is the expiration time of the signed URL in seconds
+	signedURLExpiration = 3600
+)
+
+var (
+	errInvalidFileKey = fmt.Errorf("file key must be in the format bucketID/filePath")
+)
+
+type Client struct {
+	client *supabase.Client
+}
+
+func NewClient(client *supabase.Client) *Client {
+	return &Client{
+		client: client,
+	}
+}
+
+func (c *Client) GenerateFileURL(fileKey string) (url.URL, error) {
+	bucketID, filePath := util.ExplodeFileKey(fileKey)
+	if bucketID == "" || filePath == "" {
+		return url.URL{}, fmt.Errorf("parsing file key %s: %w", fileKey, errInvalidFileKey)
+	}
+
+	resp, err := c.client.Storage.From(bucketID).CreateSignedURLForDownload(filePath, signedURLExpiration)
```

Comment:
Calling the Supabase Storage API and generating a new signed URL each time a release  is read from the database could result in significant traffic to the Storage API. 

I thing It would make sense to generate signed URLs with a longer expiration time (several hours, at least?) and cache them (using Redis for example).